src/db.rs

```diff
@@ -422,6 +433,24 @@ impl DbInner {
 
         Ok(())
     }
+
+    /// Return an error if the state has encountered
+    /// an unrecoverable error.
+    pub(crate) fn error(&self) -> Option<SlateDBError> {
```

Comment:
nit: you could have this return `Result<(), SlateDBError>` and then we could just do `let_ = self.error()?;` to check for errors.

Reply:
updated